frontend_tests/node_tests/compose_ui.js

```diff
@@ -229,3 +238,51 @@ run_test("compute_placeholder_text", () => {
     opts.private_message_recipient = "alice@zulip.com,bob@zulip.com";
     assert.equal(compose_ui.compute_placeholder_text(opts), i18n.t("Message Alice, Bob"));
 });
+
+run_test("quote_and_reply", (override) => {
+    const selected_message = {
+        type: "stream",
+        stream: "devel",
+        topic: "python",
+        sender_full_name: "Steve Stephenson",
+        sender_id: 90,
+    };
+    hash_util.by_conversation_and_time_uri = () =>
+        "https://chat.zulip.org/#narrow/stream/92-learning/topic/Tornado";
+    override(current_msg_list, "selected_message", () => selected_message);
+    override(current_msg_list, "selected_id", () => 100);
+
+    let success_function;
+    override(channel, "get", (opts) => {
+        success_function = opts.success;
+    });
+
+    $("#compose-textarea").caret = make_textbox_caret_function($("#compose-textarea"));
+
+    function set_compose_content_with_caret(content) {
+        const caret_position = content.indexOf("|");
```

Comment:
This is likely being thrown off by the mention syntax using `|` as well.  We need to either use a different symbol, or mention a different user where we won't generate the `|90` syntax?

Reply:
Changed `|` to `%`.

Reply:
That's something our test library does -- see the `zjsunit` i18n module.  We may want to turn that behavior off for this test, if it breaks what we're trying to do.